### PR TITLE
Fix v2 optimizer define order

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/PaddlePaddle/Paddle/issues/2621
Fix https://github.com/PaddlePaddle/Paddle/issues/2563

### CODE REVIEW

File: python/paddle/v2/topology.py [hunk 1]
```diff
@@ -50,6 +51,32 @@ def __check__(layers):
...
+                parameter.initial_mean = cp.g_default_initial_mean
+            if parameter.initial_std == 0.01:
+                parameter.initial_std = cp.g_default_initial_std
+            if parameter.initial_strategy == 0:
+                parameter.initial_strategy = cp.g_default_initial_strategy
+            if parameter.initial_smart == False:
+                parameter.initial_smart = cp.g_default_initial_smart
+            if parameter.num_batches_regularization == 1 and cp.g_default_num_batches_regularization:
+                parameter.num_batches_regularization = cp.g_default_num_batches_regularization
+            if parameter.gradient_clipping_threshold == 0.0 and cp.g_default_gradient_clipping_threshold:
```
Comment (lcy-seso) [inline python/paddle/v2/topology.py]: line excess 80.

File: python/paddle/v2/topology.py [hunk 2]
```diff
@@ -50,6 +51,32 @@ def __check__(layers):
 
         assert isinstance(self.__model_config__, ModelConfig)
 
+    def update_from_default(self):
+        # HACK(typhoonzero): update ParameterConfig(proto) in case of optimizers
+        # are defined after layers, or between layers.
+        # Must be called from trainer.__init__()
+        for parameter in self.__model_config__.parameters:
+            print "####", parameter.decay_rate, cp.g_default_decay_rate
```
Comment (lcy-seso) [inline python/paddle/v2/topology.py]: Is this print necessary? It seems that this is for debug print?

File: paddle/math/RowBuffer.h [hunk 3]
```diff
@@ -60,7 +60,9 @@ class RowBuffer {
    */
   inline real* get(int row) const {
     if (preallocatedBuf_) {
-      CHECK_LE((row + 1) * width_ * sizeof(real), preallocatedBuf_->getSize());
+      // CHECK_LE((row + 1) * width_ * sizeof(real),
+      // preallocatedBuf_->getSize());
+      CHECK_LE((row)*width_ * sizeof(real), preallocatedBuf_->getSize());
```
Comment (lcy-seso) [inline paddle/math/RowBuffer.h]: I am not sure what are these lines for? are these lines accidentally added?

Comment (typhoonzero) [reply paddle/math/RowBuffer.h]: Sorry, that's for my local testing, will revert this file.

File: python/paddle/v2/topology.py (hunk 1 above)
Comment (typhoonzero) [reply python/paddle/v2/topology.py]: Done.

**Pull request merged**